### PR TITLE
feat: device-level channel-management API (ADC/DIO/DAC enable, direction, output)

### PR DESCRIPTION
**Pull request by tylerkron**
Closes #241.

## Summary

Adds a device-level channel-management API on `IStreamingDevice` / `DaqifiStreamingDevice` that operates over the device's own `Channels` collection, so consumers (notably daqifi-desktop) no longer re-implement the ADC enable-bitmask encoding by hand.

### New methods
- `EnableChannel` / `EnableChannels` / `DisableChannel` / `DisableAllChannels` — maintain `IsEnabled` and encode the `ENAble:VOLTage:DC` bitmask internally over the enabled **analog** channels (the firmware treats the mask as a set-replace, so the full mask is sent each time). For **digital** channels, the global `DIO:PORt:ENAble` is toggled based on aggregate state.
- `SetDioDirection(IChannel, ChannelDirection)` → `DIO:PORt:DIRection`
- `SetDioValue(IChannel, bool)` → `DIO:PORt:STATe`
- `SetAnalogOutput(int channelNumber, double voltage)` → `SOURce:VOLTage:LEVel` + `CONFigure:DAC:UPDATE` (NQ3 only)
- `Reboot()` → `SYSTem:REboot` then `Disconnect()`

### New SCPI builders
`ScpiMessageProducer.SetAnalogOutputVoltage(int, double)` (invariant-culture decimal; rejects non-finite voltage) and `UpdateDacOutputs`.

## Design notes
- **ADC enable stays a bitmask.** Confirmed against the firmware SCPI wiki (`ENAble:VOLTage:DC 65535` = all 16 channels) and Core's existing builder.
- **Analog output is addressed by channel number.** DAC channels are not populated into `Channels` (only analog *inputs* are), so `SetAnalogOutput` takes an `int` rather than an `IChannel` — this avoids a footgun where an analog *input* channel could silently drive a DAC output.
- **Membership is enforced** for the enable/disable and DIO methods (the channel must belong to `Channels`) so internal state and the bitmask stay in sync.

## Thread-safety
The fold methods previously enumerated the live `Channels` view on the caller thread while the consumer thread can repopulate `_channels` (`Clear()`/`Add()`) on each status message — a concurrent-modification crash / torn-mask read. Repopulation is now guarded by a lock (event raised outside the lock to avoid handler deadlock), and the device-level methods fold over a locked snapshot via a new `protected SnapshotChannels()`.

## ⚠️ Breaking change
`IStreamingDevice` gains 8 new members. Any external implementer or test double must implement them (the three firmware-update test fakes in this repo were updated accordingly).

## Testing
- `dotnet build Daqifi.Core.sln`: **0 warnings, 0 errors**
- Full suite: **1058 passed, 2 skipped** (net9.0); affected suites green on net10.0.
- New tests in `DaqifiStreamingDeviceChannelManagementTests.cs` cover bitmask accumulation/set-replace, the 32-bit boundary (channel 31) and overflow guard (channel >31), single-enumeration of deferred inputs, mixed/duplicate/empty channel sets, DIO direction/value, foreign-channel + disconnected guards, analog output (incl. negative-channel and non-finite-voltage rejection), and reboot.

## Review
This PR was self-reviewed via a multi-dimension adversarial pass (correctness, edge-cases, API design, test quality, consistency, firmware/SCPI). Confirmed findings — the concurrency race, the AO signature footgun, validation ordering, DIO type-gating, non-finite voltage, and several dead-branch test gaps — were all addressed here.

## Notes / possible follow-ups
- A batch `SetAnalogOutputs(...)` overload could use the firmware's atomic multi-channel latch (each `SetAnalogOutput` call currently latches immediately); documented as immediate-apply for now.
- If `IStreamingDevice` cohesion becomes a concern, channel management could later move to capability interfaces — kept on `IStreamingDevice` here per the issue.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Daqifi.Core/Device/DaqifiStreamingDevice.cs [hunk 1]
```diff
@@ -175,6 +177,283 @@ public void StopStreaming()
...
+        /// Throws when the supplied channel is not part of this device's populated channel collection,
+        /// which would mean mutating it could not affect the device-level enable state.
+        /// </summary>
+        private void EnsureChannelBelongs(IChannel channel)
+        {
+            if (!SnapshotChannels().Contains(channel))
+            {
+                throw new ArgumentException("The specified channel does not belong to this device.", nameof(channel));
+            }
+        }
```
[the remark this answers is not in the document]
Comment (tylerkron) [reply src/Daqifi.Core/Device/DaqifiStreamingDevice.cs:456]: Fixed in 0e209c2.

**Agreed — the state-erasure half is a real bug.** `PopulateChannelsFromStatus` runs on every status message that carries port counts (init *and* any later `GetDeviceInfo` on reconnect/metadata-refresh), recreating channels with `IsEnabled=false` and silently discarding the configuration the ADC mask / DIO enable is computed from. `PopulateChannelsFromStatus` now captures the prior `(IsEnabled, Direction, OutputValue)` keyed by `(ChannelType, ChannelNumber)` before clearing and reapplies it to the freshly created instances. First-time population (no prior state) is unchanged. Added tests:
- `PopulateChannelsFromStatus_PreservesEnableDirectionAndOutputAcrossRepopulation` (unit)
- `EnableState_SurvivesStatusRefresh_AndMaskStaysCorrect` (integration: enable 0,1 → refresh → enable 2 → asserts mask `7`, not a mask that dropped 0,1)

**Declined the key-based membership suggestion (#3), intentionally.** Switching `EnsureChannelBelongs` from reference equality to `(Type, ChannelNumber)` resolution would make a *foreign* channel (e.g. `new AnalogChannel(0)`) silently accepted, which conflicts with the explicit "must belong to `Channels`" contract (and its test). With state now preserved, the live `Channels` instances carry correct state and `device.Channels` reads correctly after a refresh; the `ChannelsPopulated` event is the intended re-sync signal. A consumer caching a stale `IChannel` reference across a refresh still fails loudly with `ArgumentException` rather than silently operating on the wrong instance — an acceptable, documented contract. Happy to revisit if cached-reference reuse turns out to be a common consumer pattern.

Note: the protobuf exposes `AnalogInPortEnabled`/`DigitalPortDir`, so a future enhancement could seed state from the device's *reported* config rather than preserving in-memory state — deferred here as it depends on the firmware populating those fields reliably.
